tensorflow/core/kernels/mkl/mkl_fused_batch_norm_op.cc

```diff
@@ -548,18 +674,25 @@ class MklFusedBatchNormBwdPrimitive : public MklPrimitive {
                                       memory::format_tag::nc);
     auto mean_desc = memory::desc({1, bwdParams.depth}, MklDnnType<U>(),
                                   memory::format_tag::nc);
-    auto weights_desc = memory::desc({2, bwdParams.depth}, MklDnnType<U>(),
-                                     memory::format_tag::nc);
-    auto diff_weights_desc = weights_desc;
+#ifndef ENABLE_ONEDNN_V3
+    auto scale_shift_desc = memory::desc({2, bwdParams.depth}, MklDnnType<U>(),
+                                         memory::format_tag::nc);
+#else
+    auto scale_shift_desc =
+        memory::desc({bwdParams.depth}, MklDnnType<U>(), memory::format_tag::x);
+#endif  // !ENABLE_ONEDNN_V3
+    auto diff_scale_shift_desc = scale_shift_desc;
 
     // Forward batch-normalization descriptor and primitive descriptor.
     // Adding this back due to type difference with context.flags
-    auto bn_flags = bwdParams.training
-                        ? dnnl::normalization_flags::use_scale_shift
-                        : (dnnl::normalization_flags::use_scale_shift |
-                           dnnl::normalization_flags::use_global_stats);
+    auto bn_flags =
+        bwdParams.training
+            ? GET_SCALE_AND_SHIFT_FLAGS
+            : (GET_SCALE_AND_SHIFT_FLAGS | GET_FLAG(use_global_stats));
```

Comment:
This is the 3rd time I see this logic. Can we refactor this to an inline function, e.g., `bn_flags = GetBatchNormFlag(bwdParams.training, USE_SCALE_AND_SHIFT, use_global_stats);` ?

Reply:
Yes, fixed it.